qpc/request.py

```diff
@@ -169,58 +169,86 @@ def request(
     log_command = None
     if parser is not None:
         log_command = parser.prog
-    req_headers = {}
     token = read_client_token()
-    # create the url by adding the path to the configured server location
     url = get_server_location() + path
-    if headers:
-        req_headers.update(headers)
+    req_headers = headers or {}
     if token:
         req_headers["Authorization"] = f"Token {token}"
 
     try:
-        if method == POST:
-            result = handle_general_errors(
-                post(url, payload, req_headers), min_server_version
-            )
-        elif method == GET:
-            result = handle_general_errors(
-                get(url, params, req_headers), min_server_version
-            )
-        elif method == PATCH:
-            result = handle_general_errors(
-                patch(url, payload, req_headers), min_server_version
-            )
-        elif method == DELETE:
-            result = handle_general_errors(delete(url, req_headers), min_server_version)
-        elif method == PUT:
-            result = handle_general_errors(
-                put(url, payload, req_headers), min_server_version
+        methods = {
+            "POST": post,
+            "GET": get,
+            "PATCH": patch,
+            "DELETE": delete,
+            "PUT": put
+        }
+        try:
+            request_method = methods[method]
+            result = perform_request(
+                method,
+                request_method,
+                url,
+                params,
+                payload,
+                req_headers,
+                min_server_version
             )
-        else:
+        except KeyError:
```

Comment:
Another suggestion, though this one falls into the highly subjective debate of [EAFP](https://docs.python.org/3/glossary.html?highlight=eafp#term-EAFP) versus [LBYL](https://docs.python.org/3/glossary.html?highlight=eafp#term-LBYL).

I think I would prefer LBYL in this case because it flattens the structure and makes it easier to read. If we explicitly check for the presence of `method` first, we could remove one layer of these double-nested `try` blocks. That might look like:

```python
if method not in methods:
    logger.error("Unsupported request method %s", method)
    parser.print_help()
    sys.exit(1)

try:
    request_method = methods[method]
    result = perform_request(...)
except (requests.exceptions.ConnectionError, requests.exceptions.SSLError):
    handle_connection_error() 

log_request_info(...)
return result
```

Note that I also moved `log_request_info` and `return` outside of the `try` block since we aren't handling any exceptions that they would raise (I think).

Reply:
Alternatively, if you prefer EAFP, we could totally separate the `try`/`except` blocks like:

```python
try:
    request_method = methods[method]
except KeyError:
    logger.error("Unsupported request method %s", method)
    parser.print_help()
    sys.exit(1)

try:
    result = perform_request(...)
except (requests.exceptions.ConnectionError, requests.exceptions.SSLError):
    handle_connection_error() 

log_request_info(...)
return result
```